Tutorials/nlp components

lucasns97 opened the pull request:
Add Reader, Retriever, Chunker and translator tutorial files

lucasns97 committed:
Add translator tutorial files

Add translator tutorial files (md file and images)

lucasns97 committed:
Add chunker tutorial files

Add chunker tutorial files (md and images)

lucasns97 committed:
Add retriever files

Add retriever files (md and images)

lucasns97 committed:
Add reader files

Add reader files (md and images)

lucasns97 committed:
Update chunker and translator md files

Update chunker and translator md files

lucasns97 committed:
Add translator, chunker, reader and retriever config

Add translator, chunker, reader and retriever config to deploy.yml file

lucasns97 committed:
Update miss-spelling issues

Update miss-spelling issues on md files

lucasns97 commented:
> No tutorial chunker, sugiro a alteração "Optando por replicar os dados, para cada chunk criado a partir de um campo de texto de uma linha da tabela, todos os dados **desta** linha serão replicados de acordo com a quantidade de chunks produzidos", creio que seria a forma correta.

Muito obrigado @gvechini, fiz as alterações sugeridas

fberanizo committed:
Merge branch 'master' into tutorials/nlp-components

fberanizo committed:
Renames platiagro-components to platiagro-tasks

Ensures category is equal for all tutorials.

fberanizo committed:
Adds tags: platiagro-tasks to recently added tutorials

Ensures that filters will work for these tasks.

fberanizo commented on codelabs/platiagro-tasks/chunker.md:
```diff
@@ -0,0 +1,65 @@
+
+## Retorno esperado na implantação
+
+Espera-se como retorno numpy arrays contendo a mesma estrutura dos dados de entrada com uma coluna extra correspondente aos chunks produzidos.
```
Que tal deixar uma sugestão de exemplo de implantação (em formato JSON, como seria pra uma aplicação externa usando o modelo):

``` bash
$ curl --header "Content-Type: application/json" https://URL-DO-MODELO-IMPLANTADO -d "{"data":{"ndarray":[[1, "A bola é um objeto utilizado para lazer e em diversos desportos."]], "names": ["index", "text"]}}"

{"data":{"ndarray":[[1, "A bola é um objeto utilizado para lazer e em diversos desportos.", "A bola é um objeto utilizado para "], [1, "A bola é um objeto utilizado para lazer e em diversos desportos.", "lazer e em diversos desportos."]], "names": ["index", "text", "chunks"]}
```

lucasns97 replied:
Muito bom, vou adicionar à todos os tutoriais seguindo a mesma estratégia.

fberanizo commented on codelabs/platiagro-tasks/reader.md:
```diff
@@ -0,0 +1,84 @@
+
+1. BLEU: A métrica BLEU (cujo nome provém de BiLingual Evaluation Understudy) mede a precisão dos n-gramas das sentenças alvo geradas automaticamente em relação a um conjunto de textos de referência
+
+2. ROUGUE-L: A métrica ROUGUE (cujo nome provém de Recall-Oriented Understudy for Gisting Evaluation) identifica a co-ocorrência das substrings mais longas definidas por n-gramas entre as sentenças alvo geradas automaticamente e um conjunto de textos de referência.
```
Coloca um link para a wikipedia aqui também :)

lucasns97 replied:
Adicionado!

fberanizo commented on codelabs/platiagro-tasks/reader.md:
```diff
@@ -0,0 +1,84 @@
+
+<img src="img/reader/output_dataset_2.png" width="800">
+
+3. Uma tabela com a pontuação obtida para cada texto pelas métricas BLEU e ROUGUE-L caso seja informado a coluna de avaliação.
```
Aqui também ROUGUE -> ROUGE

lucasns97 replied:
Corrigido!

fberanizo commented on codelabs/platiagro-tasks/reader.md:
```diff
@@ -0,0 +1,84 @@
+
+## Retorno esperado na implantação
+
+Espera-se como retorno numpy arrays contendo a mesma estrutura dos dados de entrada com duas colunas extras correspondentes às respostas produzidas e suas respectivas pontuações.
```
Aqui também, um exemplo em JSON:

``` bash
$ curl --header "Content-Type: application/json" https://URL-DO-MODELO-IMPLANTADO -d "{"data":{"ndarray":[["A bola é geralmente esférica, mas pode ter outros formatos.", "Qual o formato da bola?"]], "names": ["text", "question"]}}"

{"data":{"ndarray":[["A bola é geralmente esférica, mas pode ter outros formatos.", "Qual o formato da bola?", "esférica", 0.8]], "names": ["text", "question", "answer", "answer_score"]}
```

lucasns97 replied:
Adicionado!

fberanizo commented on codelabs/platiagro-tasks/reader.md:
```diff
@@ -0,0 +1,84 @@
+Este componente utiliza um modelo de QA (Pergunta e Resposta) pré-treinado em Português com o dataset SQuAD v1.1, é um modelo de domínio público disponível em [Hugging Face](https://huggingface.co/pierreguillou/bert-large-cased-squad-v1.1-portuguese). Seu objetivo é encontrar a resposta de uma ou mais perguntas de acordo com uma lista de contextos distintos.
+
+* Neste exemplo são feitas duas perguntas distintas para dois contextos distintos.
+* As métricas computadas são o [BLEU](https://en.wikipedia.org/wiki/BLEU) e o [ROUGUE](https://en.wikipedia.org/wiki/ROUGE_(metric)).
```
Acredito que a métrica é ROUGE (e não ROUGUE).

O link para o ROUGE ficou quebrado por causa do parênteses. Você pode usar este link no lugar: https://en.wikipedia.org/wiki/ROUGE_%28metric%29

lucasns97 replied:
Corrigido!

fberanizo commented on codelabs/platiagro-tasks/retriever.md:
```diff
@@ -0,0 +1,80 @@
+
+* Neste exemplo são feitas duas perguntas distintas para dois contextos distintos.
+* A métrica computada é o [MRR](https://en.wikipedia.org/wiki/Mean_reciprocal_rank).
+* Observação: Este componente utiliza recursos da internet, portanto é importante estar conectado à rede para que este componente funcione corretamente.
```
Acho que esta observação pode ser um warning:
```
Negative
: Este componente utiliza recursos da internet, portanto é importante estar conectado à rede para que este componente funcione corretamente.
```

Visualmente fica assim:
![image](https://user-images.githubusercontent.com/1791381/140924124-4fdd7b45-c0d6-489c-9ef3-aa7481a18914.png)

lucasns97 replied:
Adicionado!

fberanizo commented on codelabs/platiagro-tasks/retriever.md:
```diff
@@ -0,0 +1,80 @@
+
+## Retorno esperado na implantação
+
+Espera-se como retorno numpy arrays contendo a mesma estrutura dos dados de entrada com duas colunas extras correspondentes às respostas produzidas e suas respectivas pontuações.
```
Aqui também, é legal colocar em formato JSON.

lucasns97 replied:
Adicionado!

fberanizo commented on codelabs/platiagro-tasks/retriever.md:
```diff
@@ -0,0 +1,80 @@
+
+As métricas de performance tem o propósito de ajudar o usuário a avaliar a performance do modelo. Essas métricas variam de acordo com o tipo de problema, tal como: classificação, regressão, agrupamento, entre outros.
+
+1. MRR: A métrica MRR (cujo nome provém de Mean Reciprocal Rank) mede a precisão da ordenação dos contextos mais prováveis de conterem a resposta de uma determinada pergunta. Comumente a notação é dada por MRR@`rank`, em que `rank` é o número N de contextos que serão elencados.
```
É legal colocar Link para a wikipedia aqui também :)

lucasns97 replied:
Adicionado!

fberanizo commented on codelabs/platiagro-tasks/translator.md:
```diff
@@ -0,0 +1,79 @@
+Utilização de modelos do tipo [MarianMT](https://huggingface.co/transformers/model_doc/marian.html) para tradução.
+
+* Neste exemplo a tradução é feita do português para o francês, mas ela pode ser feita em qualquer um dos pares de idiomas disponíveis no componente.
+* As métricas computadas são o [BLEU](https://en.wikipedia.org/wiki/BLEU) e o [ROUGUE](https://en.wikipedia.org/wiki/ROUGE_(metric)).
```
ROUGUE -> ROUGE

lucasns97 replied:
Corrigido!

fberanizo commented on codelabs/platiagro-tasks/translator.md:
```diff
@@ -0,0 +1,79 @@
+
+## Retorno esperado na implantação
+
+Espera-se como retorno numpy arrays contendo a mesma estrutura dos dados de entrada com uma coluna extra correspondente aos textos traduzidos.
```
Aqui também, um exemplo em JSON:

``` bash
$ curl --header "Content-Type: application/json" https://URL-DO-MODELO-IMPLANTADO -d "{"data":{"ndarray":[["A bola é um objeto utilizado para lazer e em diversos desportos."]], "names": ["text"]}}"

{"data":{"ndarray":[["A bola é um objeto utilizado para lazer e em diversos desportos.", "Un ballon est un objet utilisé pour les loisirs et dans divers sports."]], "names": ["text", "text_translated"]}
```

lucasns97 replied:
Adicionado!

fberanizo commented on codelabs/platiagro-tasks/translator.md:
```diff
@@ -0,0 +1,79 @@
+
+As métricas de performance tem o propósito de ajudar o usuário a avaliar a performance do modelo. Essas métricas variam de acordo com o tipo de problema, tal como: classificação, regressão, agrupamento, entre outros.
+
+1. BLEU: A métrica BLEU (cujo nome provém de BiLingual Evaluation Understudy) mede a precisão dos n-gramas das sentenças alvo geradas automaticamente em relação a um conjunto de textos de referência
+
+2. ROUGUE-L: A métrica ROUGUE (cujo nome provém de Recall-Oriented Understudy for Gisting Evaluation) identifica a co-ocorrência das substrings mais longas definidas por n-gramas entre as sentenças alvo geradas automaticamente e um conjunto de textos de referência.
```
ROUGUE->ROUGE 
Links para wiki.

lucasns97 replied:
Corrigido!

fberanizo commented on codelabs/platiagro-tasks/reader.md:
```diff
@@ -0,0 +1,84 @@
+
+* Neste exemplo são feitas duas perguntas distintas para dois contextos distintos.
+* As métricas computadas são o [BLEU](https://en.wikipedia.org/wiki/BLEU) e o [ROUGUE](https://en.wikipedia.org/wiki/ROUGE_(metric)).
+* Observação: Este componente utiliza recursos da internet, portanto é importante estar conectado à rede para que este componente funcione corretamente.
```
Acho que esta observação pode ser um warning:
```
Negative
: Este componente utiliza recursos da internet, portanto é importante estar conectado à rede para que este componente funcione corretamente.
```

Visualmente fica assim:
![image](https://user-images.githubusercontent.com/1791381/140924124-4fdd7b45-c0d6-489c-9ef3-aa7481a18914.png)

lucasns97 replied:
Adicionado!

fberanizo reviewed (changes requested):
@lucasns97 fiz alguma sugestões.

lucasns97 commented on codelabs/platiagro-tasks/chunker.md:
```diff
@@ -0,0 +1,65 @@
+tags: platiagro-tasks
+
+
+![Logotipo da PlatIAgro: possui o desenho de duas folhas verdes, uma delas é formada por linhas e pontos, como um gráfico estatístico](img/logo.png)
+
+
+# Chunker
+
```
Puts, entendi! Removido

lucasns97 commented on codelabs/platiagro-tasks/chunker.md:
```diff
@@ -0,0 +1,65 @@
+
+Separação de textos em partes menores, mantendo uma porção de sobreposição.
+
+* Neste exemplo a aplicação do Chunker é feita separando as expressões considerando as palavras, mas também pode ser realizada considerando períodos.
```
Os períodos são definidos pelo sent_tokenizer do NLTK que usualmente são divididos por ponto final. Adicionei isso à descrição e a referência da lib.

lucasns97 replied:
https://www.nltk.org/api/nltk.tokenize.html

lucasns97 committed:
Fix PR Issues

Fix PR Issues

- Misspellings
- Implemetation examples (input and output)
- Fix/add wiki urls
- Add warning for internet connection

fberanizo committed:
Merge branch 'master' into tutorials/nlp-components

fberanizo reviewed (approved):
👍
